pybamm/parameters/lithium_ion_parameters.py

```diff
@@ -848,6 +839,16 @@ def c_p_init(self, x):
         "Dimensionless initial concentration as a function of dimensionless position x"
         return self.c_p_init_dimensional(x) / self.c_p_max
 
+    def rho(self, T):
+        "Dimensionless effective volumetric heat capacity"
+        return (
+            self.rho_cn(T) * self.l_cn
+            + self.rho_n(T) * self.l_n
+            + self.rho_s(T) * self.l_s
+            + self.rho_p(T) * self.l_p
+            + self.rho_cp(T) * self.l_cp
+        ) / self.l
```

Comment:
shouldn't this take separate temperatures for neg, sep, pos?

Reply:
This only appears on the lumped models where T is the same everywhere, but I agree that it can be misleading to define it here. Maybe it would be better to define it in the model script, like we did for the `x_full`.

Reply:
ok, that's fine then